src/twisted/mail/bounce.py

```diff
@@ -75,18 +85,23 @@ def generateBounce(message, failedFrom, failedTo, transcript=''):
         'transcript': transcript,
         }
 
-    fp = StringIO.StringIO()
-    fp.write(BOUNCE_FORMAT % data)
+    fp = StringIO()
+    fp.write(BOUNCE_FORMAT.format(**data))
     orig = message.tell()
-    message.seek(2, 0)
+    message.seek(0, SEEK_END)
     sz = message.tell()
-    message.seek(0, orig)
+    message.seek(orig, SEEK_SET)
     if sz > 10000:
         while 1:
             line = message.readline()
-            if len(line)<=1:
+            if isinstance(line, bytes):
+                line = line.decode(encoding)
+            if len(line) <= 0:
```

Comment:
This logic is different from the previous logic (<=1)

Reply:
Again, this is a bugfix I found by increasing the code coverage of this module.
I found this line didn't work properly until I changed the logic to <= 0.

Reply:
The changes @rodrigc introduced actually ensure this code actually does something.  The previous `message.seek(2, 0)` ensured that `sz` would never be > 10000.  I assume the intent was `seek(0, 2)`, which is better expressed in the proposed changes.